test/optimizer/plugins/AddPathRulesOptimizer.test.ts

```diff
@@ -0,0 +1,169 @@
+      const expectedPathRule1 = new ExportablePathRule('address[1]');
+      const expectedSydneyCity = cloneDeep(sydneyCity);
+      const expectedAustraliaCountry = cloneDeep(australiaCountry);
+
+      myPackage.add(instance);
+      optimizer.optimize(myPackage);
+      expect(instance.rules).toEqual([
+        expectedPathRule0,
+        expectedTokyoCity,
+        expectedJapanCountry,
+        expectedPathRule1,
+        expectedSydneyCity,
+        expectedAustraliaCountry
+      ]);
+    });
+
+    it('should add path rules for deeper paths', () => {
+      // Instance: ChainsawPatient
+      // InstanceOf: Patient
+      // Usage: #example
+      // * address.city = "Tokyo"
+      // * address.country = "JP"
+      // * address.period.start = "2021-01-01"
+      // * address.period.end = "2021-12-31"
+      const instance = new ExportableInstance('ChainsawPatient');
+      instance.instanceOf = 'Patient';
+      instance.usage = 'Example';
+      const tokyoCity = new ExportableAssignmentRule('address.city');
+      tokyoCity.value = 'Tokyo';
+      const japanCountry = new ExportableAssignmentRule('address.country');
+      japanCountry.value = 'JP';
+      const startPeriod = new ExportableAssignmentRule('address.period.start');
+      startPeriod.value = '2021-01-01';
+      const endPeriod = new ExportableAssignmentRule('address.period.end');
+      endPeriod.value = '2021-12-31';
+      instance.rules.push(tokyoCity, japanCountry, startPeriod, endPeriod);
+
+      // Instance: ChainsawPatient
+      // InstanceOf: Patient
+      // Usage: #example
+      // * address
+      // * address.city = "Tokyo"
+      // * address.country = "JP"
+      // * address.period
+      // * address.period.start = "2021-01-01"
+      // * address.period.end = "2021-12-31"
+      const expectedPathRule = new ExportablePathRule('address');
+      const expectedTokyoCity = cloneDeep(tokyoCity);
+      const expectedJapanCountry = cloneDeep(japanCountry);
+      const expectedPeriodRule = new ExportablePathRule('address.period');
+      const expectedStartPeriod = cloneDeep(startPeriod);
+      const expectedEndPeriod = cloneDeep(endPeriod);
+
+      myPackage.add(instance);
+      optimizer.optimize(myPackage);
+      expect(instance.rules).toEqual([
+        expectedPathRule,
+        expectedTokyoCity,
+        expectedJapanCountry,
+        expectedPeriodRule,
+        expectedStartPeriod,
+        expectedEndPeriod
+      ]);
+    });
+
+    it('should add path rules for deeper paths', () => {
+      // Profile: MyPatient
+      // Parent: Patient
+      // * name MS
+      // * name.period.end TU
+      const profile = new ExportableProfile('MyPatient');
+      profile.parent = 'Patient';
+      const nameRule = new ExportableFlagRule('name');
+      nameRule.mustSupport = true;
+      const endRule = new ExportableFlagRule('name.period.end');
+      endRule.trialUse = true;
+      profile.rules.push(nameRule, endRule);
+    });
```

Comment:
It looks like this test is trying to test that path rules are included on Profiles, rather than Instances, which is good. The test description should include that, and this test needs some `expect` statements to actually test something.

Reply:
Tidied up the tests in 5df2936.